### PR TITLE
Update server monorepo to ESLint 7

### PR DESCRIPTION
**Pull request by tylerbutler**

### CODE REVIEW

**Comment by curtisman** [comment]
Close and reopening to trigger the pipeline

File: server/routerlicious/packages/services-core/src/combinedProducer.ts [hunk 1]
```diff
@@ -12,7 +12,7 @@ export class CombinedProducer implements IProducer {
     constructor(private readonly producers: IProducer[]) {
     }
 
-    public async send(messages: object[], tenantId: string, documentId: string): Promise<any> {
+    public async send(messages: Record<string, unknown>[], tenantId: string, documentId: string): Promise<any> {
```
Comment (tanviraumi) [inline server/routerlicious/packages/services-core/src/combinedProducer.ts:15]: Is this what typescript recommends now? Converting an object to Record<K, T> for better typing?

Comment (tanviraumi) [reply server/routerlicious/packages/services-core/src/combinedProducer.ts:15]: Also this implements IProducer so I am wondering why TS is not complaining? Are objects and Record<string, unknown> interchangable?

**Review by tanviraumi — APPROVED** [review]

File: server/routerlicious/packages/services-core/src/combinedProducer.ts (hunk 1 above)
Comment (tanviraumi) [reply server/routerlicious/packages/services-core/src/combinedProducer.ts:15]: And there are more implementations of IProducer. So may be change them as well?

Comment (tylerbutler) [reply server/routerlicious/packages/services-core/src/combinedProducer.ts:15]: This was a mistake. I experimented with some fixes and got sloppy about keeping them separate from the ESLint 7 upgrade itself. I'll remove these and replace with the disable comment. If we want to make this or a similar change we should do it separately from this PR.

Comment (tylerbutler) [reply server/routerlicious/packages/services-core/src/combinedProducer.ts:15]: Resolved in an upcoming push.

File: server/routerlicious/packages/services-utils/src/auth.ts [hunk 2]
```diff
@@ -16,7 +16,7 @@ export function validateTokenClaims(
     documentId: string,
     tenantId: string,
     maxTokenLifetimeSec: number,
-    isTokenExpiryEnabled: boolean): ITokenClaims {
+    isTokenExpiryEnabled: boolean): ITokenClaims | undefined {
```
Comment (tylerbutler) [inline server/routerlicious/packages/services-utils/src/auth.ts:19]: I don't understand how this compiled before...

**Pull request merged**